File: python/36 Fibonacci.py
def fib(num):
    if num <= 0:
        return
    elif num == 1:
        print(0)
        return
    prev = 0
    current = 1
    print(prev, current, end=" ")
    for _ in range(num - 2):
        prev, current = current, current + prev
        print(current, end=" ")
num = int(input("Enter a number: "))
fib(num)

File: java/36 Fibonacci.java
package code;
import java.util.Scanner;
public class Test {
	static void fib(int num) {
		if (num <= 0) {
            return;
        } else if (num == 1) {
            System.out.print("0");
            return;
        }
		int prev = 0;
        int current = 1;
        System.out.print(prev + " " + current + " ");
        for(int i=2;i<num;i++) {
        	int next=prev+current;
        	System.out.print(next+" ");
        	prev=current;
        	current=next;
        }
	}
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter a nummber: ");
		int num1=sc.nextInt();
		fib(10);
	}
}
